File: foodguide/src/main/java/com/vinhkhanh/foodguide/entity/Food.java
package com.vinhkhanh.foodguide.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "food")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Food {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "food_id")
    private Long foodId;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "price", nullable = false)
    private Double price;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "location_id", nullable = false)
    private Location location;
}

File: vinh-khanh-food-guide/shared/models/Food.ts
/**
 * Shared Food Model - Compatible with Spring Boot Backend
 */

export interface Food {
  foodId: number;
  name: string;
  price: number;
  description?: string;
  locationId: number;
}

export interface FoodRequest {
  name: string;
  price: number;
  description?: string;
  locationId: number;
}
